metadata_layer.py

```diff
@@ -574,56 +574,32 @@ def finalize(self, service, service_type, collection, start=NO_VALUE,
             self.collection_id = None
         else:
             self.collection_id = collection.id
-        if self.start is self.NO_VALUE:
+        if self.start is None:
             self.start = start
-        if self.finish is self.NO_VALUE:
-            if finish is self.NO_VALUE:
+        if self.finish is None:
+            if finish is None:
                 finish = datetime.datetime.utcnow()
             self.finish = finish
-        if self.start is self.NO_VALUE:
+        if self.start is None:
             self.start = self.finish
-        if self.counter is self.NO_VALUE:
+        if self.counter is None:
             self.counter = counter
-        if self.exception is self.NO_VALUE:
+        if self.exception is None:
             self.exception = exception
 
     def collection(self, _db):
         return get_one(_db, Collection, id=self.collection_id)
 
     def apply(self, _db):
-        if any(x is self.NO_VALUE for x in [self.service, self.service_type,
-                                            self.collection_id]):
+        if any(x is None for x in [self.service, self.service_type]):
```

Comment:
When we were using NO_VALUE to mean "value has not been set" it made sense to check collection_id. Now that we're using None, it's ambiguous whether `collection_id=None` means "collection_id set to None" or "collection_id has not been set", but I think it's not too bad -- you're supposed to call `finalize`, which takes `collection` as a required argument, so you have to explicitly set it to `None`.